packages/agent-builder/src/workflows/template-builder.ts

```diff
@@ -0,0 +1,1682 @@
+        copiedFiles,
+        conflicts,
+        message,
+      };
+    } catch (error) {
+      console.error('Programmatic file copy failed:', error);
+
+      return {
+        success: false,
+        copiedFiles: [],
+        conflicts: [],
+        message: `Programmatic file copy failed: ${error instanceof Error ? error.message : String(error)}`,
+        error: error instanceof Error ? error.message : String(error),
+      };
+    }
+  },
+});
+
+// Step 9: Intelligent merging with AgentBuilder
+const intelligentMergeStep = createStep({
+  id: 'intelligent-merge',
+  description: 'Use AgentBuilder to intelligently merge template files',
+  inputSchema: IntelligentMergeInputSchema,
+  outputSchema: IntelligentMergeResultSchema,
+  execute: async ({ inputData, runtimeContext }) => {
+    console.log('Intelligent merge step starting...');
+    const { conflicts, copiedFiles, commitSha, slug, templateDir, branchName } = inputData;
+    const targetPath = inputData.targetPath || runtimeContext.get('targetPath') || process.cwd();
+    try {
+      // Create copyFile tool for edge cases
+      const copyFileTool = createTool({
+        id: 'copy-file',
+        description:
+          'Copy a file from template to target project (use only for edge cases - most files are already copied programmatically).',
+        inputSchema: z.object({
+          sourcePath: z.string().describe('Path to the source file relative to template directory'),
+          destinationPath: z.string().describe('Path to the destination file relative to target project'),
+        }),
+        outputSchema: z.object({
+          success: z.boolean(),
+          message: z.string(),
+          error: z.string().optional(),
+        }),
+        execute: async ({ context }) => {
+          try {
+            const { sourcePath, destinationPath } = context;
+
+            // Use templateDir directly from input
+            const resolvedSourcePath = resolve(templateDir, sourcePath);
+            const resolvedDestinationPath = resolve(targetPath, destinationPath);
+
+            if (existsSync(resolvedSourcePath) && !existsSync(dirname(resolvedDestinationPath))) {
+              await mkdir(dirname(resolvedDestinationPath), { recursive: true });
+            }
+
+            await copyFile(resolvedSourcePath, resolvedDestinationPath);
+            return {
+              success: true,
+              message: `Successfully copied file from ${sourcePath} to ${destinationPath}`,
+            };
+          } catch (error) {
+            return {
+              success: false,
+              message: `Failed to copy file: ${error instanceof Error ? error.message : String(error)}`,
+              error: error instanceof Error ? error.message : String(error),
+            };
+          }
+        },
+      });
+
+      // Initialize AgentBuilder for merge and registration
+      const agentBuilder = new AgentBuilder({
+        projectPath: targetPath,
+        mode: 'template',
```

Comment:
I'm not sold that "mode" is the way we want to go with this, it should be easier to customize